[Neo Jia Han] iP

jia-han opened the pull request:
# Duke the chatbot
- Text based
- Fast to use

How to start Duke?
1. Download the .jar file here: https://github.com/jia-han/ip/releases
2. Run it!
3. Enjoy :smile:

Features:
- [x] Keeps track of different kinds of tasks
- [x] Simple search function
- [ ] Reminders (*coming soon*)

For programmers who want the text-based interface instead, here is the `main` and `run` method to switch to: 
```java
private void run() {
        ui.showWelcome();
        taskList = new TaskList(storage.read());
        while (true) {
            try {
                Command command = parser.parse(ui.readCommand());
                command.execute(storage, ui, taskList);
            } catch (DukeException e) {
                ui.showError(e);
            }
        }

public static void main(String[] args) {
        Duke duke = new Duke();
        duke.run();
    }
```

> "Quack." - A duck

Commit:
Add Gradle support

jia-han committed:
Add greet, echo and exit functions

jia-han committed:
Add storing of text

jia-han committed:
Add marking of tasks

jia-han committed:
Add subtasks

jia-han committed:
I/O text testing

jia-han committed:
Add exceptions

jia-han committed:
Add delete for tasks and more exceptions

jia-han committed:
Add testing for level 5 and 6

jia-han committed:
Add saving functionality for tasks

jia-han committed:
Add parsing of strings to dateTime and update automated test

jia-han committed:
Merge branch 'branch-Level-7'

jia-han committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

jia-han committed:
Add fix for reading of duke.txt file when loading the program, and update automated text testing

jia-han committed:
Refactor Duke.java and update automated testing

jia-han committed:
Refactor duke files into correspoding packages

jia-han committed:
Add gradle support

jia-han committed:
Add JUnit tests for DeleteCommand and Parser

jia-han committed:
Update mainClassName for jar executable to run properly

jia-han committed:
Add JavaDoc

jia-han committed:
Add coding standards to Gradle

jia-han committed:
Add find function

jia-han committed:
Merge branch 'branch-A-CodingStandard'

jia-han committed:
Merge branch 'branch-Level-9'

jia-han committed:
Update to fit coding standards

jia-han committed:
Add automating checkstyle

ngshijun commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,45 @@
+    /**
+     *  Constructs a specified Command task for Duke that does not require any Duke utility classes.
+     */
+    public Command() {
```
Perhaps this constructor is not necessary since Java will automatically make one for you if the desired constructor is as empty constructor.

ngshijun commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,38 @@
+ * Created when user inputs "todo".
+ */
+public class TodoCommand extends Command {
+    private String taskDetails;
```
```suggestion
    private String taskDetail;
```

ngshijun commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,38 @@
+ */
+public class Task {
+    private String task;
+    private boolean done;
```
```suggestion
    private boolean isDone;
```

ngshijun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,42 @@
+    private Parser parser = new Parser();
+    private TaskList taskList;
+
+    public Duke() {
```
Empty constructor will be provided by Java automatically.

ngshijun commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,43 @@
+ * Created when user inputs "event".
+ */
+public class EventCommand extends Command {
+    private String taskDetails;
```
```suggestion
    private String taskDetail;
```

ngshijun commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,42 @@
+ * Created when user inputs "deadline".
+ */
+public class DeadlineCommand extends Command {
+    private String taskDetails;
```
```suggestion
    private String taskDetail;
```

ngshijun commented on src/main/java/duke/exception/DateTimeException.java:
```diff
@@ -0,0 +1,14 @@
+    @Override
+    public String toString() {
+        return super.toString() + "Please use the specified date-time format: yyyy-MM-dd HH:mm,"
+            + " or yyyy-MM-dd if you want the time to be 23:59";
```
Indentation for wrapped lines should be 8 spaces

ngshijun commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,123 @@
+                            split[1] = split[1] + " 23:59";
+                        }
+                        LocalDateTime time = LocalDateTime.parse(split[1].substring(3),
+                            DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));
```
Indentation for wrapped lines should be 8 spaces

ngshijun commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,122 @@
+            break;
+        case 'D':
+            LocalDateTime time = LocalDateTime.parse(event[1].substring(0, event[1].indexOf(")")),
+                DateTimeFormatter.ofPattern("dd-MMM-yyyy HH:mm"));
```
Indentation for wrapped lines should be 8 spaces

ngshijun commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,122 @@
+            break;
+        case 'E':
+            time = LocalDateTime.parse(event[1].substring(0, event[1].indexOf(")")),
+                DateTimeFormatter.ofPattern("dd-MMM-yyyy HH:mm"));
```
Indentation for wrapped lines should be 8 spaces

ngshijun reviewed (commented):
Overall coding standard is good.

jia-han committed:
Add GUI to Duke

jia-han committed:
Merge branch 'branch-Level-10'

jia-han committed:
Merge branch 'branch-A-Checkstyle'

Shawn532 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,39 @@
+    public Duke() {
+    }
+
+    /**
+     * duke.gui.Main entry function.
+     * @param args Command line arguments.
+     */
```
Nitpicking here, but I think there should be a blank line between the description and the param section.

Shawn532 reviewed (commented):
Couldn't find much code quality errors. Code seems very clean and Javadocs for relevant methods are consise.

jia-han committed:
Add edit function

Users are unable to edit tasks that they have input previously

Edit function allows for changing of task description so they do not
have to delete and create new task

Let's add the edit function

jia-han committed:
Add assertions

Checks that will never be reached are being done.

Assertions will reduce the number of redundant checks.

Let's update the code to replace the redundant checks and ensure that
assumptions held are correct.

jia-han committed:
Add null checks and readability fixes

Duke class: Add javadocs for getResponse method
Parser class: Refactor parsing of commands for readability
MainWindow class: Add check for null images

jia-han committed:
Merge pull request #3 from jia-han/branch-C-Update

Add edit function

jia-han committed:
Merge branch 'master' into branch-A-CodeQuality

jia-han committed:
Merge pull request #2 from jia-han/branch-A-CodeQuality

Add null checks and readability fixes

jia-han committed:
Merge branch 'master' into branch-A-Assertions

jia-han committed:
Merge pull request #1 from jia-han/branch-A-Assertions

Add assertions

jia-han committed:
Update README

jia-han committed:
Fix parsing for edit command
Update boolean values to fit coding standards

jia-han committed:
Add welcome message
Add product screenshot
Fix automatic resizing of GUI

jia-han committed:
Change version number

jia-han committed:
Fix for jar executable not working
